src/librustc_passes/diagnostics.rs

```diff
@@ -261,6 +261,19 @@ let result = loop { // ok!
 ```
 "##,
 
+E0642: r##"
+Trait methods currently cannot take patterns as arguments.
+
+Example of erroneous code:
+
+```compile_fail,E0642
+trait Foo {
+    fn foo((x, y): (i32, i32)); // error: patterns aren't allowed
+                                // in methods without bodies
+}
+```
+"##,
```

Comment:
Last nitpick, could you expand the documentation to show the "fixed" code, using maybe `x_and_y` as argument name?